src/app.py

```diff
@@ -72,8 +73,11 @@ def name_page():
         # Don't call the backend API unless we have something.
         html = ""
     else:
-        html = dataset.name_lookup(metadata, **request.args)
-    entities = api.get_query_fields(metadata)
+        entities = api.get_query_fields(metadata)
+        html = dataset.name_lookup(metadata, entities, **request.args)
+
+    if entities is None:
+        entities = api.get_query_fields(metadata)
```

Comment:
Maybe we could have this line run on L67 unconditionally, and we could alter `get_query_fields` so it accepts `None` and returns an empty list in that case?

Reply:
Done in b776691